force-app/main/default/classes/SOQL.cls

```diff
@@ -39,9 +41,11 @@ public with sharing class SOQL {
      * @return A list of Contact records.
      * Update the corresponding test class and testGetContactsByFirstName method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static List<Contact> getAllContacts (List<String> contactsbyFirstName) {
+        String filterBy = 'John';
+        List<Contact> allContacts = [SELECT Id, Name FROM Contact WHERE FirstName = :filterBy];
+        return allContacts;
```

Comment:
You need to actually use the argument in line 44 in your query: what's the point of having an argument you pass into a method if you aren't going to use it? Remove the filterBy parameter as that makes the method useless.

Reply:
Or, remove the argument.